### PR TITLE
fix(#2143): allow env var overrides for js query server config

### LINKED ISSUE #2143: `couchdb` and `couchdb.cmd` startup scripts do not allow overriding JS/CoffeeScript variables
## Description

It should be possible for a user to set `COUCHDB_QUERY_SERVER_JAVASCRIPT` or `COUCHDB_QUERY_SERVER_COFFEESCRIPT` or the Fauxton document root in the shell before invoking `/opt/couchdb/bin/couchdb` or `couchdb.cmd` and have those settings carry over to CouchDB.

Unfortunately, we forcibly override those settings:

https://github.com/apache/couchdb/blob/57bf82e5cc3c3ab55ad6c3e46ea2c666ac2711cc/rel/files/couchdb.in#L29-L31

https://github.com/apache/couchdb/blob/c850ffa889dc1f8bf562107aa07da429357bc8ef/rel/files/couchdb.cmd.in#L28-L29

## Steps to Reproduce

Export the environment var, then invoke `couchdb` or `couchdb.cmd`.

## Suggested fix:

Use the following structure:

```bash
export COUCHDB_QUERY_SERVER_JAVASCRIPT=${COUCHDB_QUERY_SERVER_JAVASCRIPT:-{{prefix}}/bin/couchjs {{prefix}}/share/server/main.js"
```

or, in the `couchdb.cmd` file:

```batch
IF NOT DEFINED COUCHDB_QUERY_SERVER_JAVASCRIPT SET COUCHDB_QUERY_SERVER_JAVASCRIPT="{{prefix}}/bin/couchjs {{prefix}}/share/server/main.js"
```

## Your Environment

* CouchDB Version used: 2.3.1

## Additional context

Migrated from https://github.com/apache/couchdb-docker/issues/149

/cc @janl

### LINKED ISSUE #2404: 3.0 build is broken on Microsoft Windows
Should be something simple.

```
==> config (compile)
==> b64url (compile)
==> ets_lru (compile)
==> khash (compile)
==> snappy (compile)
==> ioq (compile)
==> hqueue (compile)
Compiling c:/relax/couchdb/src/hqueue/c_src/hqueue.c
Microsoft (R) C/C++ Optimizing Compiler Version 18.00.40629 for x64
Copyright (C) Microsoft Corporation.  All rights reserved.

cl : Command line error D8021 : invalid numeric argument '/Werror'
ERROR: compile failed while processing c:/relax/couchdb/src/hqueue: rebar_abort
make: *** [couch] Error 1
```

### PR DESCRIPTION
**Pull request by janl**
Closes #2143
Fixes #2404

### CODE REVIEW

File: rel/files/couchdb.in [hunk 1]
```diff
@@ -26,8 +26,8 @@ export BINDIR="$ROOTDIR/erts-$ERTS_VSN/bin"
 export EMU=beam
 export PROGNAME=`echo $0 | sed 's/.*\///'`
 
-export COUCHDB_QUERY_SERVER_JAVASCRIPT="{{prefix}}/bin/couchjs {{prefix}}/share/server/main.js"
-export COUCHDB_QUERY_SERVER_COFFEESCRIPT="{{prefix}}/bin/couchjs {{prefix}}/share/server/main-coffee.js"
+export COUCHDB_QUERY_SERVER_JAVASCRIPT=${COUCHDB_QUERY_SERVER_JAVASCRIPT:-{{prefix}}/bin/couchjs {{prefix}}/share/server/main.js"
```
Comment (wohali) [inline rel/files/couchdb.in]: Unbalanced quotation marks

File: rel/files/couchdb.cmd.in [hunk 2]
```diff
@@ -25,8 +25,9 @@ set EMU=beam
 set PROGNAME=%~n0
 set PATH=%PATH%;%COUCHDB_BIN_DIR%
 
-set COUCHDB_QUERY_SERVER_JAVASCRIPT="{{prefix}}/bin/couchjs {{prefix}}/share/server/main.js"
-set COUCHDB_QUERY_SERVER_COFFEESCRIPT="{{prefix}}/bin/couchjs {{prefix}}/share/server/main-coffee.js"
+IF NOT DEFINED COUCHDB_QUERY_SERVER_JAVASCRIPT SET COUCHDB_QUERY_SERVER_JAVASCRIPT="{{prefix}}/bin/couchjs {{prefix}}/share/server/main.js"
```
Comment (wohali) [inline rel/files/couchdb.cmd.in]: This should work but needs empirical validation, I can do this Monday

**Comment by janl** [comment]
fixed in latest push

File: rel/files/couchdb.in (hunk 1 above)
Comment (janl) [reply rel/files/couchdb.in]: fixed

File: rel/files/couchdb.cmd.in (hunk 2 above)
Comment (janl) [reply rel/files/couchdb.cmd.in]: Thanks!

**Comment by wohali** [comment]
The CouchDB build is broken on Windows; I need to debug that first before I can validate this change.

**Comment by wohali** [comment]
Will test this tomorrow when i'm back near Windows.

File: rel/files/couchdb.cmd.in (hunk 2 above)
Comment (wohali) [reply rel/files/couchdb.cmd.in]: You need to remove the quotation marks:

```batch
 IF NOT DEFINED COUCHDB_QUERY_SERVER_JAVASCRIPT SET COUCHDB_QUERY_SERVER_JAVASCRIPT={{prefix}}/bin/couchjs {{prefix}}/share/server/main.js
```

**Comment by wohali** [comment]
This will fix #2404.

**Comment by janl** [comment]
nice

**Pull request merged**